shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -38,40 +39,47 @@ private void extractResources() {
                 deleteFiles();
             }
 
-            final AssetManager manager = mContext.getResources().getAssets();
+            try (ZipFile apk = new ZipFile(mContext.getPackageCodePath())) {
+                byte[] buffer = null;
+                for (String asset : mResources) {
+                    try {
+                        final File output = new File(dataDir, asset);
 
-            byte[] buffer = null;
-            for (String asset : mResources) {
-                try {
-                    final File output = new File(dataDir, asset);
-
-                    if (output.exists()) {
-                        continue;
-                    }
-                    if (output.getParentFile() != null) {
-                        output.getParentFile().mkdirs();
-                    }
-
-                    try (InputStream is = manager.open(asset)) {
-                        try (OutputStream os = new FileOutputStream(output)) {
-                            if (buffer == null) {
-                                buffer = new byte[BUFFER_SIZE];
-                            }
+                        if (output.exists()) {
+                            continue;
+                        }
+                        if (output.getParentFile() != null) {
+                            output.getParentFile().mkdirs();
+                        }
 
-                            int count = 0;
-                            while ((count = is.read(buffer, 0, BUFFER_SIZE)) != -1) {
-                                os.write(buffer, 0, count);
+                        ZipEntry entry = apk.getEntry("assets/" + asset);
```

Comment:
I believe this directory name is configurable and not always guaranteed to the called "assets".

Reply:
https://android.googlesource.com/platform/frameworks/base/+/master/libs/androidfw/AssetManager.cpp#438
The `kAssetsRoot` is constant value defined https://android.googlesource.com/platform/frameworks/base/+/master/libs/androidfw/AssetManager.cpp#62 . Android doesn't provide any chance to change the value.